### PR TITLE
epic: esl media v2

### PR DESCRIPTION
**Pull request by ala-n**
*ESL Media* 
- cleanup and extend API
- fix dynamic property change
- add autodetection for provider

### CODE REVIEW

**Comment by ala-n** [comment]
recheckcla

**Comment by ala-n** [comment]
recheckcla

**Comment by ala-n** [comment]
recheckcla

**Comment by ala-n** [comment]
recheckcla

**Comment by ala-n** [comment]
recheckcla

**Comment by ala-n** [comment]
recheckcla

**Comment by ala-n** [comment]
recheckcla

File: jest.config.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ module.exports = {
   ],
   collectCoverageFrom: [
     'src/modules/**/*.ts',
+    // test dir
+    '!src/**/test/*ts',
```
Comment (dshovchko) [inline jest.config.js:17]: Missing comma

File: src/modules/esl-media/providers/brightcove-provider.ts [hunk 2]
```diff
@@ -17,17 +17,26 @@ export interface BCPlayerAccount {
...
 @BaseProvider.register
 export class BrightcoveProvider extends BaseProvider {
-  static get providerName() {
-    return 'brightcove';
-  }
-
-  static videojsClasses = 'video-js vjs-default-skin video-js-brightcove';
+  static readonly providerName = 'brightcove';
+
+  // static readonly providerRegexp = /(?:http(?:s)?:\/\/)(?:www\.)?players.brightcove/;
```
Comment (dshovchko) [inline src/modules/esl-media/providers/brightcove-provider.ts:24]: Commented code

**Review by dshovchko — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Comment by ala-n** [comment]
recheckcla

**Review by nattallius — APPROVED** [review]

**Comment by nattallius** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ala-n** [comment]
recheckcla

**Pull request closed without merging**